store/src/java/com/zimbra/cs/mailbox/MailboxManager.java

```diff
@@ -888,19 +890,28 @@ private synchronized Mailbox createMailboxInternal(OperationContext octxt, Accou
                     instantiateExternalVirtualMailbox(data) : instantiateMailbox(data);
             mbox.setGalSyncMailbox(isGalSyncAccount);
             // the existing Connection is used for the rest of this transaction...
-            mbox.beginTransaction("createMailbox", octxt, redoRecorder, conn);
-
-            if (created) {
-                // create the default folders
-                mbox.initialize();
-            }
+            try (final MailboxLock l = mbox.lock(true);
+                 final Mailbox.MailboxTransaction t = mbox.new MailboxTransaction("createMailbox", octxt,l,redoRecorder, conn)) {
+                if (created) {
+                    // create the default folders
+                    mbox.initialize();
+                }
 
-            // cache the accountID-to-mailboxID and mailboxID-to-Mailbox relationships
-            cacheAccount(data.accountId, data.id);
-            cacheMailbox(mbox);
-            redoRecorder.setMailboxId(mbox.getId());
+                // cache the accountID-to-mailboxID and mailboxID-to-Mailbox relationships
+                cacheAccount(data.accountId, data.id);
+                cacheMailbox(mbox);
+                redoRecorder.setMailboxId(mbox.getId());
 
-            success = true;
+                try {
+                    if (mbox != null) {
```

Comment:
If `mbox == null` you are going to throw an NPE in the `try-with-resources` head. I think the logic here is wrong. Why was this moved from the finally block below? The control flow was changed. We do not want to do that.